### PR TITLE
Earth atmosphere computations to core

### PR DESCRIPTION
**Pull request by Yash-10**
The `jacchia` module had a lot of computations inside the high-level module. Here, the attempt is to just move `jacchia` and some more computations to `core`.

- I am wondering if there would already be an in-built method that does the equivalent of `_get_index`
- It would have been great if `_check_altitude` method could also be moved to `core` but the f-strings in the `ValueError` condition seemed to prohibit it.

Thanks!

### CODE REVIEW

**Review by astrojuanlu — APPROVED** [review]
Looks good to me! Documentation is failing because of this:

```
sphinx.errors.SphinxWarning: Cannot resolve import of unknown module poliastro.core.earth_atmosphere.util in poliastro.earth.atmosphere.base
```

File: src/poliastro/earth/atmosphere/base.py [hunk 1]
```diff
@@ -62,17 +68,20 @@ def _check_altitude(self, alt, r0, geometric=True):
...
             z = h_to_z(h, r0)
 
+        z, h = z * u.km, h * u.km
+
         # Assert in range
-        if z < self.zb_levels[0] or z > self.zb_levels[-1]:
+        if not self.zb_levels[0] <= z <= self.zb_levels[-1]:
             raise ValueError(
                 f"Geometric altitude must be in range [{self.zb_levels[0]}, {self.zb_levels[-1]}]"
             )
```
[the remark this answers is not in the document]
Comment (Yash-10) [reply src/poliastro/earth/atmosphere/base.py:87]: Getting rid of that dynamic error message means converting it into a "non-dynamic" error message that just says "Invalid range", right?

[the remark this answers is not in the document]
Comment (astrojuanlu) [reply src/poliastro/earth/atmosphere/base.py:87]: Converting it to a non-dynamic error message, that doesn't depend on the variables

[the remark this answers is not in the document]
Comment (astrojuanlu) [reply src/poliastro/earth/atmosphere/base.py:87]: In any case, I think it's good to keep the error message as it is now 👍🏽

**Comment by astrojuanlu** [comment]
Merging, thanks @Yash-10 !

File: src/poliastro/earth/atmosphere/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Yash-10) [reply src/poliastro/earth/atmosphere/base.py:87]: I think we could move `_check_altitude` computation to core and still have the dynamic error message in the high-level module @astrojuanlu

[the remark this answers is not in the document]
Comment (astrojuanlu) [reply src/poliastro/earth/atmosphere/base.py:87]: Woops, good catch! Opened https://github.com/poliastro/poliastro/issues/1281 for that

**Pull request merged**